mlir/test/Target/Cpp/no_extra_semicolon.mlir

```diff
@@ -0,0 +1,18 @@
+// RUN: mlir-translate -mlir-to-cpp %s | FileCheck %s
+// RUN: mlir-translate -mlir-to-cpp -declare-variables-at-top %s | FileCheck %s
+
+func.func @no_extra_semicolon(%arg0: i1) {
+  emitc.if %arg0 {
+    emitc.if %arg0 {
+    }
+    emitc.verbatim "return;"
+  }
```

Comment:
Can we check the missing semicolon for the other emitc ops that are handled, like `DeclareFuncOp` `ForOp`, `SwitchOp` and `IncludeOp`?
This doesn't need to be here, it can also be in their respective tests.

I think you need to take extra care with the check lines as they will look for a matching prefix, so will also pass if the line has something else at the end (like a semicolon). I.e. `// CHECK-NEXT: if ([[V0]]) {` will also match `if (v0) {;`
I think there is FileCheck command line args or syntax to enforce full line matching.

Reply:
The point though is how you write them, since accounting for all the permutations of where the for/switch/etc is is a bit noisy, so I wrote in a more of a regression test style, where I've noticed the issue.

I'll add a full line matching in the tests where those are used already, so it'll probably help with regressions.


>I think there is FileCheck command line args or syntax to enforce full line matching.

Thanks, I completely forgot about it, have to pass the match full lines flag.

Reply:
Updated.